code/modules/warhammer_cults/blood_cult/objects.dm

```diff
@@ -0,0 +1,140 @@
+							explosion(1, 1, 1, 1, 1)
+							fire_stacks = max(300, fire_stacks)
+							IgniteMob()
+			switch(choice)
+				if("YES")
+					to_chat(src, "<span='notice'>You took the right choice. Welcome to the cult, the party is just starting!</span>")
+					mind.god = KHORNE
+				if("NO")
+					to_chat(src, "<span='warning'>You are ignorant. But He had mercy upon you, continue your futile struggles.</span>")
+
+/obj/structure/chaos/skull_totem
+	name = "Skull Totem"
+	desc = "A totem with a skull attached on top, you feel some bizarre presence emanating from it."
+	icon = 'icons/obj/blood_cult.dmi'
+	icon_state = "totem"
+	density = 1
+	anchored = 1
+	var/mob/player
+	var/active = FALSE
+
+	attack_hand(mob/user)
+		if(user.mind.god == KHORNE)
+			if(!active)
+				if(!player)
+					to_chat(user, "You throw a drop of blood inside the skull totem, imprinting it with your own soul.")
+					player = user
+				else
+					to_chat(player, "You feel uncomfortable for a second.")
+					to_chat(user, "You remove the blood inside the skull totem, erasing the soul mark it had imprinted.")
+					player = null
+			else
+				to_chat(user, "You cant imprint your soul in the totem if it is active.")
+		else
+			to_chat(user, "<span='warning'> YOU AREN'T A SERVANT OF KHORNE! DONT TOUCH IT!")
+
+	proc/activate()
+		var/mob/living/carbon/human/M = player
+		active = TRUE
+		icon_state = "[icon_state]-active"
+		light_power = 10
+		light_range = 3
+		light_color = LIGHT_COLOR_FLARE
+		sleep(100)
+		M.mind.god.give_gifts(M)
+		deactivate()
+
+	proc/deactivate()
+		explosion(loc,0,0,0,0)
+		qdel(src)
+
+/obj/structure/reagent_dispensers/khorne_altar
+	name = "Fountain of the Blood God"
+	desc = "This altar works as a direct connection with the ruinous powers."
+	icon = 'icons/obj/blood_cult.dmi'
+	icon_state = "blood_altar"
+	breakable = FALSE
+	density = 1
+	anchored = 1
+	amount_per_transfer_from_this = 10
+	possible_transfer_amounts = "10;25;50;100"
+	initial_capacity = INFINITY
+	initial_reagent_types = list(/datum/reagent/daemonic_blood = 1)
+	light_power = 10
+	light_range = 3
+	light_color = LIGHT_COLOR_FLARE
+	var/active = FALSE
+
+	attack_hand(mob/living/carbon/human/user)
+		if(user.mind.god == KHORNE)
+			if(active != TRUE)
+				start_ritual(user)
+				active = TRUE
+				to_chat(user, "<span='warning'>You start the blood ritual.</span>")
+
+	proc/start_ritual(mob/living/carbon/human/user)
+		icon_state = "[icon_state]-active"
+		var/list/totems
+		for(var/obj/structure/chaos/skull_totem/S in orange(3, src))
+			totems += S																	//FOR SOME FUCKING REASON IT GETS STUCK HERE
+			if(totems.len >= 4) //minimum totems = 4
+				for(S in totems)
+					S.activate()
+					var/obj/item/projectile/beam/blood_effect/effect = new(get_turf(src))
+					effect.pixel_x = 0
+					effect.pixel_y = 0
+					effect.launch_projectile(totems)
+					spawn(100)
+						icon_state = "blood_altar"
+						active = FALSE
```

Comment:
Why is active = FALSE for both conditions of the if statement?

Reply:
Because on both cases, the altar or fountain is deactivating, forced or finished the ritual